### ISSUE TITLE
Multiple problems with route handlers

### ISSUE THREAD

**Issue by cdoremus** [uki00a/fresh-testing-library#38]
I have been trying to get `fresh-testing-library` route handlers to work. Here are the three things that are holding me back:
1. The TypeScript typing of the manifest file is broken
2. Properties of the data object are not defined.
3. Stack traces point to the wrong file:

Please see the  code below and advise what is wrong with it.

---
For example. here is one route I am working on:

### About route code
```ts
import { HandlerContext, Handlers, PageProps } from "$fresh/server.ts";
import { CSS, render } from "gfm";

const markdownFile = "About.md";
const baseUrl = Deno.env.get("IS_PROD")
  ? "https://deno-blog.deno.dev"
  : "http://localhost:8000";

export type AboutHandlerData = {
  blog: string;
}
export const handler: Handlers = {
  async GET(_: Request, ctx: HandlerContext<AboutHandlerData>) {
    const fileName = markdownFile;
    const path = `${Deno.cwd()}/static/${fileName}`;
    const contents = await Deno.readTextFile(path);
    const blog = render(contents, { baseUrl });
    return ctx.render({ blog });
  },
};

export default function AboutPage({ data }: PageProps) {
  const { blog } = data;
  return (
    <div class="p-4 mx-auto max-w-screen-md">
      <style dangerouslySetInnerHTML={{ __html: CSS }} />
      <div
        class="bg-white p-5 markdown-body"
        data-color-mode="light"
        data-light-theme="light"
        data-dark-theme="dark"
        dangerouslySetInnerHTML={{ __html: blog }}
     >
        {blog}
      </div>
    </div>
  );
}
```
### About route handler test code:
```ts
import { createHandlerContext } from "$fresh-testing-library";
import { describe, it } from "std/testing/bdd.ts";
import { handler} from "../../routes/about.tsx";
import manifest from "../../fresh.gen.ts";
import { assert, assertEquals } from "std/assert/mod.ts";

describe("routes/About.tsx handler", () => {

  it("should respond to GET request...s", async () => {
    assert(handler.GET);
    const req = new Request("http://localhost:8000/abosut");
    const ctx = createHandlerContext(req, { manifest });
    ctx.state.data =  {blog: ""};

    const res = await handler.GET(req, ctx);
    assertEquals(res.status, 200);

  })
});
```
### Manifest typing errors
```sh
TS2322 [ERROR]: Type '{ routes: { "./routes/[name].tsx": typeof $0; "./routes/_404.tsx": typeof $1; "./routes/_app.tsx": typeof $2; "./routes/_middleware.ts": typeof $3; "./routes/about.tsx": typeof $4; "./routes/credits.tsx": typeof $5; "./routes/index.tsx": typeof $6; }; islands: { ...; }; baseUrl: string; }' is not assignable to type 'Manifest'.
  Types of property 'routes' are incompatible.
    Type '{ "./routes/[name].tsx": typeof $0; "./routes/_404.tsx": typeof $1; "./routes/_app.tsx": typeof $2; "./routes/_middleware.ts": typeof $3; "./routes/about.tsx": typeof $4; "./routes/credits.tsx": typeof $5; "./routes/index.tsx": typeof $6; }' is not assignable to type 'Record<string, RouteModule | MiddlewareModule<any> | AppModule | ErrorPageModule | UnknownPageModule>'.
      Property '"./routes/_middleware.ts"' is incompatible with index signature.
        Type 'typeof import("file:///home/craig/git/deno/blog/deno-blog/routes/_middleware.ts")' is not assignable to type 'RouteModule | MiddlewareModule<any> | AppModule | ErrorPageModule | UnknownPageModule'.
          Type 'typeof import("file:///home/craig/git/deno/blog/deno-blog/routes/_middleware.ts")' is not assignable to type 'UnknownPageModule'.
            Types of property 'handler' are incompatible.
              Type '(_req: Request, _ctx: MiddlewareHandlerContext<Record<string, unknown>>) => (_req: Request, ctx: MiddlewareHandlerContext<Record<string, unknown>>) => Promise<...>' is not assignable to type 'UnknownHandler'.
                Types of parameters '_ctx' and 'ctx' are incompatible.
                  Type 'UnknownHandlerContext<Record<string, unknown>>' is missing the following properties from type 'MiddlewareHandlerContext<Record<string, unknown>>': next, destination, params
    const ctx = createHandlerContext(req, { manifest });
                                            ~~~~~~~~
    at file:///home/craig/git/deno/blog/deno-blog/tests/routes/about.test.tsx:14:45

Found 2 errors.
```

### Data object properties not defined with stack trace pointing to the wrong code
___Note that the stack tract points to the [name].tsx file and not about.tsx___
```sh
craig@craig-G750JX:~/git/deno/blog/deno-blog$ deno test -A --no-check tests/routes/about.test.tsx 
running 1 test from ./tests/routes/about.test.tsx
routes/About.tsx handler ...
  should respond to GET request. ... FAILED (32ms)
routes/About.tsx handler ... FAILED (due to 1 failed step) (34ms)

 ERRORS 

routes/About.tsx handler ... should respond to GET request...s => https://deno.land/std@0.201.0/testing/_test_suite.ts:323:15
error: TypeError: Cannot destructure property 'blog' of 'data' as it is undefined.
  const { blog } = data;
          ^
    at BlogPostPage (file:///home/craig/git/deno/blog/deno-blog/routes/[name].tsx:28:11)
    at renderRouteComponent (https://deno.land/x/fresh_testing_library@0.8.0/internal/fresh/mod.ts:132:24)
    at Object.render (https://deno.land/x/fresh_testing_library@0.8.0/server.ts:120:18)

 FAILURES 

routes/About.tsx handler ... should respond to GET request...s => https://deno.land/std@0.201.0/testing/_test_suite.ts:323:15

FAILED | 0 passed | 1 failed (1 step) (609ms)

error: Test failed
```

**Comment by cdoremus** [comment]
Here is my test code on Github: https://github.com/cdoremus/deno-blog/blob/fresh-testing-lib/tests/routes/about.test.tsx

**Comment by uki00a** [comment]
@cdoremus Thanks for reporting the issue!

For the first problem, the following patch to [routes/_middleware.ts](https://github.com/cdoremus/deno-blog/blob/dedc81e58be5647170fb086fba9e98d192dc09ca/routes/_middleware.ts) should solve it:

```diff
-export const handler(
-  _req: Request,
-  _ctx: MiddlewareHandlerContext,
-) {
-  return setCacheControlHeaders();
-}
+export const handler = setCacheControlHeaders();
```

The second problem should be resolved in [v0.8.1](https://github.com/uki00a/fresh-testing-library/releases/tag/0.8.1).
